src/PageHeader/PageHeader.tsx

```diff
@@ -0,0 +1,393 @@
+  children,
+  hidden = hiddenOnRegularAndWide,
+  sx = {},
+}) => {
+  const isHidden = useResponsiveValue(hidden, false)
+  const contentNavStyles = {
+    display: isHidden ? 'none' : 'flex',
+    flexDirection: 'row',
+    alignItems: 'center',
+    gap: '0.5rem',
+    order: REGION_ORDER.ContextArea,
+  }
+  return <Box sx={merge<BetterSystemStyleObject>(contentNavStyles, sx)}>{children}</Box>
+}
+type LinkProps = Pick<
+  React.AnchorHTMLAttributes<HTMLAnchorElement>,
+  'download' | 'href' | 'hrefLang' | 'media' | 'ping' | 'rel' | 'target' | 'type' | 'referrerPolicy'
+>
+export type ParentLinkProps = PageHeaderProps & LinkProps
+
+const ParentLink = React.forwardRef<HTMLAnchorElement, ParentLinkProps>(
+  (
+    {
+      children,
+      sx = {},
+      href,
+      'aria-label': ariaLabel = `Back to ${children}`,
+      as = 'a',
+      hidden = hiddenOnRegularAndWide,
+    },
+    ref,
+  ) => {
+    const isHidden = useResponsiveValue(hidden, false)
+    return (
+      <>
+        <Link
+          ref={ref}
+          as={as}
+          aria-label={ariaLabel}
+          muted
+          sx={merge<BetterSystemStyleObject>(
+            {
+              display: isHidden ? 'none' : 'flex',
+              alignItems: 'center',
+              lineHeight: '2.75rem', // min touch target size 44px (16*2.75)
+              gap: '0.5rem',
+            },
+            sx,
+          )}
+          href={href}
+        >
+          <ArrowLeftIcon />
+          <Box>{children}</Box>
+        </Link>
+      </>
+    )
+  },
+) as PolymorphicForwardRefComponent<'a', ParentLinkProps>
+
+// ContextBar
+// Generic slot for any component above the title region. Use it for custom breadcrumbs and other navigation elements instead of ParentLink.
+// ---------------------------------------------------------------------
+
+const ContextBar: React.FC<React.PropsWithChildren<PageHeaderProps>> = ({
+  children,
+  sx = {},
+  hidden = hiddenOnRegularAndWide,
+}) => {
+  const isHidden = useResponsiveValue(hidden, false)
+  return <Box sx={merge<BetterSystemStyleObject>({display: isHidden ? 'none' : 'flex'}, sx)}>{children}</Box>
+}
+
+// ContextAreaActions
+// ---------------------------------------------------------------------
+const ContextAreaActions: React.FC<React.PropsWithChildren<PageHeaderProps>> = ({
+  children,
+  sx = {},
+  hidden = hiddenOnRegularAndWide,
+}) => {
+  const isHidden = useResponsiveValue(hidden, false)
+  return (
+    <Box
+      sx={merge<BetterSystemStyleObject>(
+        {
+          display: isHidden ? 'none' : 'flex',
+          flexDirection: 'row',
+          alignItems: 'center',
+          gap: '0.5rem',
+          flexGrow: '1',
+          justifyContent: 'right',
+        },
+        sx,
+      )}
+    >
+      {children}
+    </Box>
+  )
+}
+
+const MEDIUM_TITLE_HEIGHT = '2rem'
+const LARGE_TITLE_HEIGHT = '3rem'
+
+const TitleAreaContext = React.createContext<{
```

Comment:
I cannot seem to reply your [comment](https://github.com/primer/react/pull/2265#discussion_r1042809668) so I'll write here.

>Did you choose 2rem based on the title font size and line height? 

Yes! 🙌🏼 

>If so, I'd store 2rem in a variable and use that variable for setting the heights

Great idea, done!

>Also - we'll need to change 2rem to match the font size

This is a great catch! Sorry I missed that. I thought about how to solve this and tried moving the `variant` prop to `TitleArea` (parent of `Title`) from `Title`. I think lifting the state helped both technically.

I leveraged using React context and easily shared the data between children. Creating a React context for the TitleArea and `variant` into the context and use this context in each child (`LeadingAction`,`LeadingAction`,`TrailingVisual`,`TrailingAction`) to manage children's height. 

I think moving `variant` to `TitleArea` feels more intuitive to me. Looking forward to hearing what you think @mperrotti! 

There is one thing that concerns me though and it is the sizes of the visuals or icon buttons. Should we increase their font-sizes too? 🤔

Reply:
> There is one thing that concerns me though and it is the sizes of the visuals or icon buttons. Should we increase their font-sizes too? 🤔

I don't think anything should affect the button size except the button's `size